hostinfo/hostinfo_freebsd.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build freebsd
+// +build freebsd
+
+package hostinfo
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"strings"
+
+	"golang.org/x/sys/unix"
+	"tailscale.com/util/lineread"
+	"tailscale.com/version/distro"
+)
+
+func init() {
+	osVersion = osVersionFreebsd
+}
+
+func osVersionFreebsd() string {
+	dist := distro.Get()
+	propFile := "/etc/os-release"
+
+	// FreeBSD 12.2 added /etc/os-release, earlier versions did not have it.
+	// Be aware: pfSense 2.5.2 made /etc/os-release a symlink pointing nowhere.
+	m := map[string]string{}
+	lineread.File(propFile, func(line []byte) error {
+		eq := bytes.IndexByte(line, '=')
+		if eq == -1 {
+			return nil
+		}
+		k, v := string(line[:eq]), strings.Trim(string(line[eq+1:]), `"'`)
+		m[k] = v
+		return nil
+	})
+
+	un := unix.Utsname{}
+	unix.Uname(&un)
+
+	var attrBuf strings.Builder
+	attrBuf.WriteString("; version=")
+	for _, b := range un.Release {
+		if b == 0 {
+			break
+		}
+		attrBuf.WriteByte(byte(b))
+	}
+	attr := attrBuf.String()
+
+	version := "FreeBSD"
+	switch dist {
+	case distro.Pfsense:
+		b, _ := ioutil.ReadFile("/etc/version")
+		version = fmt.Sprintf("pfSense %s", b)
+	case distro.OPNsense:
+		b, err := exec.Command("opnsense-version").Output()
+		if err == nil {
+			version = string(b)
+		} else {
+			version = "OPNsense"
+		}
+	case distro.TrueNAS:
+		b, err := ioutil.ReadFile("/etc/version")
```

Comment:
just `os.ReadFile` these days. ioutil is now deprecated.

Reply:
Done.